[Eliana Pang] iP

eeliana opened the pull request:
# Duke v0.1
> I have not failed. I’ve just found 10,000 ways that won’t work. – Thomas Edison [source](https://criteriaforsuccess.com/18-awesome-technology-quotes-to-inspire-motivate/)

Duke v0.1 frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* FAST SUPER FAST

All you need to do is,
1. download it from [here](https://github.com/eeliana/ip).
2. double-click it.
2. add your tasks.
4. let it manage your tasks for you

And it is FREE!
Features:
- [x]  Managing tasks
- [ ]  Managing deadlines (in progress)
- [ ]  Reminders (coming soon)

Here is the main method:
```java
public static void main(String[] args) {
    String directoryPath = "./data";
    String filePath = "./data/duke.text";

    File directory = new File(directoryPath);
    // Check folder exists
    if (!directory.exists()) {
        boolean created = directory.mkdirs();
        if (created) {
            System.out.println("Successfully created directory.");
        } else {
            System.out.println("An error occurred");
        }
    }

    Duke duke = new Duke(new TaskList(), new Storage(filePath), new Ui());
    duke.run();
}
```

Commit:
Add Gradle support

eeliana committed:
Added greet, echo, exit features

eeliana committed:
Abstracted greet, echo, exit to functions

eeliana committed:
Added classes ListItem and List to keep track of the user's tasks

eeliana committed:
Added add and list functionality to Duke

eeliana committed:
Implemented mark as done functionality

eeliana committed:
Added and implemented 3 types of tasks and updated Duke.

eeliana committed:
Tested with automating testing for text UIs

eeliana committed:
Handle invalid inputs and empty task description

eeliana committed:
Implemented delete task functionality and some javadocs

eeliana committed:
Added javadocs for all classes

eeliana committed:
Added enums for fixed commands by Duke.

eeliana committed:
Update exception handling in Duke

eeliana committed:
Remove syntax errors

eeliana committed:
Update Duke's methods to non-class-level methods

eeliana committed:
Add FileManager to manage file operations

eeliana committed:
Fix bug replacing string when marking task as completed.

eeliana committed:
Add DateTimeManager to parse date strings

eeliana committed:
Add command to print deadlines/events occurring on a specific date.

eeliana committed:
Rename FileManager to Storage

eeliana committed:
Add Ui to deal with interactions with user

eeliana committed:
Add nested class Parser to interpret user commands

eeliana committed:
Put classes into packages

eeliana committed:
Update master branch

eeliana committed:
Merge branch 'master' of https://github.com/eeliana/ip

eeliana committed:
Merge branch 'branch-Level-7' of https://github.com/eeliana/ip into branch-Level-7

eeliana committed:
Merge branch 'master' into branch-Level-7

eeliana committed:
Update spacing

eeliana committed:
Merge branch 'master' into branch-Level-7

eeliana committed:
Update spacing

eeliana committed:
Merge branch 'branch-Level-7'

eeliana committed:
Add javadov for method returnTasksOnDate

eeliana committed:
Update javadocs

eeliana committed:
Update spacing

eeliana committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/duke/DukeException.java

eeliana committed:
Fix syntax errors.

eeliana committed:
Add testing for DateTimeManager and Storage classes

eeliana committed:
Update text-ui-testing

eeliana committed:
Update layout of files to coding standards.

eeliana committed:
Update JavaDocs for Duke class and the various tasks classes.

eeliana committed:
Update JavaDocs for all classes

eeliana committed:
Add JavaDocs for public methods in Storage class.

eeliana committed:
Add JavaDocs for public methods in DateTimeManager.

eeliana committed:
Update JavaDocs in Task and TaskList classes.

eeliana committed:
Update JavaDocs for Duke class.

eeliana committed:
Initialise uninitialised variables in Storage according to coding standards.

eeliana committed:
Initalise uninitialised variables in Duke.

eeliana committed:
Update JavaDocs for spacing.

eeliana committed:
Add find functionality for user to search tasks by keyword

eeliana committed:
Merge branch 'branch-A-JavaDoc'

eeliana committed:
Merge branch 'branch-A-CodingStandard'

eeliana committed:
Merge branch 'branch-Level-9'

eeliana committed:
Add Manifest file for JAR.

eeliana committed:
Refactor Duke's commands to be encapsulated in a Command class

Commands were previously in an enum. Each type of user command is now encapsulated
in individual classes which extends from a Command superclass.

This allow details such as the required actions for each command to be abstracted
from Duke.

eeliana committed:
Add HelpCommand for usage help

eeliana commented:
# Duke v0.1
> I have not failed. I’ve just found 10,000 ways that won’t work. – Thomas Edison [Source] (https://criteriaforsuccess.com/18-awesome-technology-quotes-to-inspire-motivate/)

Duke v0.1 frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* FAST SUPER FAST

All you need to do is,
1. download it from [here](https://github.com/eeliana/ip).
2. double-click it.
2. add your tasks.
4. let it manage your tasks for you

And it is FREE!
Features:
- [x]  Managing tasks
- [ ]  Managing deadlines (in progress)
- [ ]  Reminders (coming soon)

Here is the main method:
```java
public static void main(String[] args) {
    String directoryPath = "./data";
    String filePath = "./data/duke.text";

    File directory = new File(directoryPath);
    // Check folder exists
    if (!directory.exists()) {
        boolean created = directory.mkdirs();
        if (created) {
            System.out.println("Successfully created directory.");
        } else {
            System.out.println("An error occurred");
        }
    }

    Duke duke = new Duke(new TaskList(), new Storage(filePath), new Ui());
    duke.run();
}
```

eeliana committed:
Merge branch 'add-gradle-support' of https://github.com/eeliana/ip into add-gradle-support

eeliana committed:
Update README.md

eeliana committed:
Fix bugs in running the different commands

eeliana committed:
Configure Checkstyle rules and update files to follow the rules

eeliana committed:
Refactor UI-related code to its own class

eeliana committed:
Refactor Duke GUI

eeliana committed:
Fix styling errors

eeliana committed:
Merge branch 'branch-Level-10'

angkoonhwee commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,277 @@
+            } else if (userInput.equals("list")) {
+                return new ListCommand(ui, taskList);
+            } else if (userInput.equals("help")) {
+                return new HelpCommand(ui, taskList);
```
Nice help feature! 💪

angkoonhwee commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,277 @@
+
+            this.listOfWords = arrOfCommandWords;
+            // Check the command word
+            String commandWord = arrOfCommandWords[0];
```
Maybe can group line 63 and 64 together as a block then the comment line 63 can be placed before the block? And maybe a break after line 64 to separate the block from the switch statement?

angkoonhwee commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,277 @@
+            // Check the command word
+            String commandWord = arrOfCommandWords[0];
+            switch (commandWord) {
+            case "todo":
```
Why not make these commands a constant encapsulated in the task object so when there is a change to command word it will be easier?

eeliana replied:
Hello, can you elaborate on what you mean by this? Do you mean using enums in the Task class or I should maybe separate the parsing say Task.parse(...) which returns a task command and maybe a TaskList.parse(...) for "done" and "delete"?

angkoonhwee replied:
yes something like that because according to the textbook, they say there shouldn't be "magic strings/numbers"

eeliana replied:
Thank you! I have tried refactoring, not sure what to do with the magic strings though.

angkoonhwee commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,277 @@
+        private int taskIndex = -1;
+        private LocalDate date = LocalDate.now();
+
+        private Command parseString(String userInput) throws DukeException {
```
Maybe condense or split the function into smaller chunks since the coding guidelines recommend to avoid long methods (less than 30 LOC) [reference](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)

angkoonhwee commented on src/main/java/seedu/duke/command/Command.java:
```diff
@@ -0,0 +1,53 @@
+        return false;
+    }
+
+    public boolean updatesTaskList() {
```
perhaps can make the variable name sound more boolean like 'isUpdateTaskList'?

eeliana committed:
Refactor Parser functions

nicolej2122 reviewed (commented):
Looks good! Adheres to basic code quality guidelines

eeliana committed:
Add asserts at important points in the code to document assumptions

eeliana committed:
Refactor code to get rid of "magic" literals

Parsing user input has hardcoded strings as switch cases. These strings
represents command types which appear as magic literals.

Using an enum to parse and handle these cases improves readability by
making the cases clearer.

eeliana committed:
Add reminders for deadline and event tasks

eeliana committed:
Update GUI to include reminder function

Reminder command lists out reminders in a task list format as with the
other commands.

GUI is refactored to allow user to click on a button to get reminder on
upcoming tasks, with the dates sorted in ascending order. The earliest
task is at the top.

User can easily view reminders with the earliest one being at the top of the
reminder list.

eeliana committed:
Minor refactoring of code for alert

eeliana committed:
Add Undo command

Allow user to undo commands made during the current session of Duke.

Ability to undo command can be convenient to not have to delete the task
and add it back again.

eeliana committed:
Merge pull request #1 from eeliana/branch-A-Assertions

Add asserts at important points in the code to document assumptions

eeliana committed:
Merge branch 'master' into branch-A-CodeQuality

eeliana committed:
Merge pull request #2 from eeliana/branch-A-CodeQuality

Refactor code to get rid of "magic" literals

eeliana committed:
Merge pull request #4 from eeliana/branch-B-Reminders

Update GUI to include reminder function

eeliana committed:
Merge pull request #3 from eeliana/branch-C-Undo

Add Undo functionality to Duke

eeliana committed:
Refactor Storage class to according to code quality

eeliana committed:
Update profile pictures of duke and user

eeliana committed:
Fix bugs in existing commands

eeliana committed:
Pop reminder alert on duke start up

eeliana committed:
Add Ui.png

eeliana committed:
Update README.md

eeliana committed:
Update Javadocs

eeliana committed:
Set theme jekyll-theme-hacker

eeliana committed:
Set theme jekyll-theme-cayman

eeliana committed:
Set theme jekyll-theme-cayman

eeliana committed:
Update README.md

eeliana committed:
Update README.md 

Add an extra line between header and table to fix github markdown error in github pages

eeliana committed:
Update _config.yml

Change markdown parser

eeliana committed:
Update Ui.png

eeliana committed:
Fix bugs with executing Duke's commands

eeliana committed:
Merge branch 'master' of https://github.com/eeliana/ip

eeliana committed:
Refactor Ui.png to correct folder

eeliana committed:
Create JAR file for Duke

eeliana committed:
Update hash map of dates and tasks after delete command

eeliana committed:
Remove Undo Command due to bugs

eeliana committed:
Update DateTimeManager parsing format

eeliana committed:
Fix adding deadline/event without date bug

eeliana committed:
Fix bug with delete function

eeliana committed:
Update files to follow checkstyle

eeliana committed:
Update build.gradle file
